src/irmin-pack/s.ml

```diff
@@ -55,41 +55,92 @@ module type Specifics = sig
+
+    (** {2 Low-level API} *)
 
-      If [unlink] is false then temporary files and files from the previous
-      generation will be kept on disk after the gc finished. This option is
-      useful for debugging. The default is true.
+    val start_exn :
+      ?unlink:bool -> throttle:throttle -> repo -> commit_key -> bool Lwt.t
+    (** [start_exn] tries to start the gc process and returns true if the gc is
+        launched.
 
-      If [throttle] is [Skip] and there is a concurrent gc, [start_gc] returns
-      false immediately without launching a second gc. If [throttle] is [Block]
-      and there is a concurrent gc, [start_gc] blocks waiting for the previous
-      gc to finish and only after launches a second gc. If the previous GC
-      failed, the function returns without launching a new GC.
+        If [unlink] is false then temporary files and files from the previous
+        generation will be kept on disk after the gc finished. This option is
+        useful for debugging. The default is true.
 
-      TODO: Detail exceptions raised. *)
+        If [throttle] is [Skip] and there is a concurrent gc, [start_exn]
+        returns false immediately without launching a second gc. If [throttle]
+        is [Block] and there is a concurrent gc, [start_exn] blocks waiting for
+        the previous gc to finish and only after launches a second gc. If the
+        previous GC failed, the function returns without launching a new GC.
 
-  val finalise_gc : ?wait:bool -> repo -> bool Lwt.t
-  (** [finalise_gc ?wait repo] waits for the gc process to finish in order to
-      finalise it. It returns true if a GC was finalised.
+        TODO: Detail exceptions raised. *)
 
-      Finalising consists of mutating [repo] so that it points to the new file
-      and to flush the internal caches that could be referencing GCed objects.
+    val finalise_exn : ?wait:bool -> repo -> bool Lwt.t
+    (** [finalise_exn ?wait repo] waits for the gc process to finish in order to
+        finalise it. It returns true if a GC was finalised.
 
-      If [wait = true] (the default), the call blocks until the GC process
-      finishes. If [wait = false] it either returns [false], raises an exception
-      or finalises and returns [true].
+        Finalising consists of mutating [repo] so that it points to the new file
+        and to flush the internal caches that could be referencing GCed objects.
 
-      If there are no running gcs, the call is a no-op and it returns false.
+        If [wait = true] (the default), the call blocks until the GC process
+        finishes. If [wait = false] it either returns [false], raises an
+        exception or finalises and returns [true].
+
+        If there are no running gcs, the call is a no-op and it returns false.
+
+        TODO: Detail exceptions raised. *)
+
+    (** {2 High-level API} *)
+
+    val run :
+      ?finished:(float -> unit) ->
+      repo ->
+      commit_key ->
+      (bool, string) result Lwt.t
```

Comment:
The usual way to return string as errors is to wrap them into a `Msg of string` variant so that they could be extended later on without breaking the API too much. See https://erratique.ch/software/rresult/doc/Rresult/index.html#usage for a nice overview of how to design error APIs :-)